File: src/main/java/com/example/hbs/psi/impl/HbPartialNameImpl.java
package com.example.hbs.psi.impl;

import com.example.hbs.psi.HbPartialName;
import com.intellij.lang.ASTNode;
import org.jetbrains.annotations.NotNull;

public class HbPartialNameImpl extends HbPsiElementImpl implements HbPartialName {
  public HbPartialNameImpl(@NotNull ASTNode astNode) {
    super(astNode);
  }

  @Override
  public String getName() {
    return getText();
  }
}

File: src/main/kotlin/com/example/hbs/psi/impl/HbPartialNameImpl.kt
package com.example.hbs.psi.impl

import com.example.hbs.psi.HbPartialName
import com.intellij.lang.ASTNode

class HbPartialNameImpl(node: ASTNode) : HbPsiElementImpl(node), HbPartialName {

    override val name: String
        get() = text
}
